File: packages/java/src/main/java/fbc/Seguranca/ResultadoAtaque.java
package fbc.Seguranca;

import java.util.Map;

/**
 * Resultado de rodar um ataque contra um alvo.
 *
 * {@code vulneravel = true} significa que o ataque ACHOU um problema (a cifra
 * falhou). {@code vulneravel = false} significa que a cifra resistiu a esse
 * ataque específico.
 */
public class ResultadoAtaque {
    private final String nomeAtaque;
    private final boolean vulneravel;
    private final Severidade severidade;
    private final String detalhes;
    private final Map<String, Object> dados;

    public ResultadoAtaque(String nomeAtaque, boolean vulneravel, Severidade severidade, String detalhes) {
        this(nomeAtaque, vulneravel, severidade, detalhes, null);
    }

    public ResultadoAtaque(
            String nomeAtaque,
            boolean vulneravel,
            Severidade severidade,
            String detalhes,
            Map<String, Object> dados) {
        this.nomeAtaque = nomeAtaque;
        this.vulneravel = vulneravel;
        this.severidade = severidade;
        this.detalhes = detalhes;
        this.dados = dados;
    }

    public String nomeAtaque() {
        return nomeAtaque;
    }

    public boolean vulneravel() {
        return vulneravel;
    }

    public Severidade severidade() {
        return severidade;
    }

    public String detalhes() {
        return detalhes;
    }

    public Map<String, Object> dados() {
        return dados;
    }

    public String linhaResumo() {
        String status;
        if (severidade == Severidade.PULADO) {
            status = "PULADO";
        } else if (severidade == Severidade.DEMONSTRACAO) {
            status = "DEMONSTRAÇÃO";
        } else {
            status = vulneravel ? "❌ VULNERÁVEL" : "✅ resistiu";
        }

        return "[" + status + "] " + nomeAtaque + " (" + severidade + "): " + detalhes;
    }
}

File: packages/go/internal/seguranca/resultado_ataque.go
package seguranca

import "fmt"

// Severidade classifica o resultado de um ataque.
type Severidade string

const (
	SevCritica      Severidade = "critica"
	SevAlta         Severidade = "alta"
	SevMedia        Severidade = "media"
	SevBaixa        Severidade = "baixa"
	SevInfo         Severidade = "info"
	SevPulado       Severidade = "pulado"
	SevDemonstracao Severidade = "demonstracao"
	SevErro         Severidade = "erro"
)

// ResultadoAtaque é o resultado de rodar um ataque contra um alvo.
//
// Vulneravel = true significa que o ataque ACHOU um problema (a cifra falhou).
// Vulneravel = false significa que a cifra resistiu a esse ataque específico.
type ResultadoAtaque struct {
	NomeAtaque string
	Vulneravel bool
	Severidade Severidade
	Detalhes   string
	Dados      map[string]any
}

// LinhaResumo formata o resultado para o relatório da suíte.
func (r ResultadoAtaque) LinhaResumo() string {
	var status string
	switch r.Severidade {
	case SevPulado:
		status = "PULADO"
	case SevDemonstracao:
		status = "DEMONSTRAÇÃO"
	default:
		if r.Vulneravel {
			status = "❌ VULNERÁVEL"
		} else {
			status = "✅ resistiu"
		}
	}

	return fmt.Sprintf("[%s] %s (%s): %s", status, r.NomeAtaque, r.Severidade, r.Detalhes)
}
